### PR TITLE
Ditch pipecat and fix hangs

### PR DESCRIPTION
**Pull request by DemiMarie**
Instead of using named pipes and an extra subprocess, just use the stdin
and stdout of qrexec-client-vm.  This avoids a subprocess and prevents
the client hanging forever if the connection is refused.

Fixes QubesOS/qubes-issues#7233

### CODE REVIEW

File: src/gpg-client.c [hunk 1]
```diff
@@ -93,77 +73,34 @@ int main(int argc, char *argv[])
...
 
     switch (pid = fork()) {
         case -1:
             perror("fork");
             exit(1);
         case 0:
-            devnull = open("/dev/null", O_RDONLY);
-            if (devnull < 0) {
-                perror("open /dev/null");
+            if (dup2(pipe_in[0], 0) == -1 || dup2(pipe_out[1], 1) == -1) {
```
[the remark this answers is not in the document]
Comment (DemiMarie) [reply src/gpg-client.c]: All of those file descriptors are marked close-on-exec, due to the use of the `O_CLOEXEC` flag to `pipe2()`.  Therefore, they will be closed automatically by the kernel during `execve()`.

[the remark this answers is not in the document]
Comment (marmarek) [reply src/gpg-client.c]: Ah, right.

**Comment by DemiMarie** [comment]
PipelineRetry

**Comment by DemiMarie** [comment]
PipelineRetry

**Review by marmarek — APPROVED** [review]

**Pull request merged**